src/test/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActorTest.scala

```diff
@@ -152,14 +152,43 @@ class TaskLauncherActorTest extends MarathonSpec with GivenWhenThen {
       .setOperator(Operator.UNIQUE)
       .setValue("")
       .build
+    val unreachableStrategy = UnreachableEnabled(5.minutes, 10.minutes)
+    val constraintApp: AppDefinition = f.app.copy(constraints = Set(uniqueConstraint), unreachableStrategy = unreachableStrategy)
+    val offer = MarathonTestHelper.makeBasicOffer().build()
+
+    val lostInstance = TestInstanceBuilder.newBuilder(f.app.id).addTaskUnreachable(unreachableStrategy = unreachableStrategy).getInstance()
+
+    Mockito.when(instanceTracker.instancesBySpecSync).thenReturn(InstanceTracker.InstancesBySpec.forInstances(lostInstance))
+    val captor = ArgumentCaptor.forClass(classOf[InstanceOpFactory.Request])
+    // we're only interested in capturing the argument, so return value doesn't matter
+    Mockito.when(instanceOpFactory.matchOfferRequest(captor.capture())).thenReturn(f.noMatchResult)
+
+    val launcherRef = createLauncherRef(instances = 1, constraintApp)
+    launcherRef ! RateLimiterActor.DelayUpdate(constraintApp, clock.now())
+
+    val promise = Promise[MatchedInstanceOps]
```

Comment:
Where do we use this?

Reply:
good question, actually the old api accepting promises ... I`ll remove the promise :) thanks!